src/wp-includes/class-wp-block-type.php

```diff
@@ -353,6 +367,10 @@ public function __get( $name ) {
 	 *              or false otherwise.
 	 */
 	public function __isset( $name ) {
+		if ( 'variations' === $name ) {
+			return true;
```

Comment:
No, this is running into an error for some older logic. `null !== $this->variations` is the problem.
The issue is with the default logic,  isset logic was returning `false` as the default for `$variations` was `null`,  however, this contradicts to trunk code as it returns `true` as the default for `$variaitons` was `array()`.

At present to handle B/C for default value, we return `array()` if anyone tries to fetch variations directly. 
But due to `isset` logic we were getting different values for `$variations` than expected at certain places 
For e.g this place here - `https://github.com/10up/wordpress-develop/blob/9225964063a0609de92268fea28d0aed66d86542/src/wp-admin/includes/post.php#L2323`
See this error - https://github.com/WordPress/wordpress-develop/actions/runs/7461317691/job/20301216384?pr=5718#step:14:340

This is the reason why I had to return true always, which is kind of correct according to the new logic. 
See the code snippet below. 
```php
// This does set variations to null
$block_type->variations = null;

// However if I fetch variations immediately I'll get a blank array. 
$value =  $block_type->variations;  // $block_type->get_variations();
print_r( $value ); // this will print `array()`

// so isset must return true as practically variations can never be null, atleast for external entity
```

This opens up a new issue, which is we can never set `$variations` to null and run external logic based on a null value.

Now the open question is - 
Do we need the default value as `array()` or do we need the ability to set `variations` to `null`. It looks like we can't have both privileges.

Reply:
Thanks for the explanation, @kt-12. This is definitely unintuitive, but does make sense based on the current expected default value for $block_type->variations being an empty array. I can't think of a use case where someone would want to manually set that property to a null value after it was instantiated, and that would likely break things anyway since the property is documented to be an array always, so I think what you're proposing is fine.

Reply:
Thanks for the feedback. Seems right to be.